modules/Bio/EnsEMBL/Variation/DBSQL/VCFCollectionAdaptor.pm

```diff
@@ -208,6 +210,17 @@ sub new {
       }
     }
 
+    # Check that the remote/local VCF file exists before creating a corresponding VCFCollection object
+    my $filename_template = $hash->{filename_template} =~ /(nfs|ftp:)/ ? $hash->{filename_template} : $root_dir.$hash->{filename_template};
+
+    if ($filename_template !~ /[#]+[^#]+[#]+/) {
+      my $file_exists = ($hash->{type} eq 'remote') ? $self->_ftp_file_exists($filename_template) : (-e $filename_template);
+      if (!$file_exists) {
+        warn("WARNING: Can't access to the ".$hash->{species}." VCF file '$filename_template' (".$hash->{id}.")");
+        next;
+      }
+    }
```

Comment:
The issue is it failed badly if we continue the code: creating a VCFCollection object with a non existing file

Reply:
I was wondering if there is anything else to do if the if-statement doesn't evaluate to true.

Reply:
Sorry I get it now! You can only test the ones that don't have a hash in their name.

Reply:
Well if the file doesn't exist or is not accessible, there is not much else to do.
I prefered to use a `warning` to let the user know that there is an issue and to avoid killing all the processes (i.e. when we want to display all the pop frequencies on the website for a given variant)